src/robotFactory.js

```diff
@@ -1,10 +1,78 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, { x = 0, y = 0 }, chipVersion) {
```

Comment:
just use a default value, like this:
```suggestion
  constructor(name, weight, { x = 0, y = 0 } = {}, chipVersion) {
```